core/modules/dashboard/dashboard.module

```diff
@@ -0,0 +1,404 @@
+    'access dashboard' => array(
+      'title' => t('Access Dashboard'),
+      'description' => t('Access the dashboard.'),
+    ),
+    'administer dashboard' => array(
+      'title' => t('Aminister Dashboard'),
+      'description' => t('Access the dashboard settings.'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_config_info().
+ */
+function dashboard_config_info() {
+  $prefixes['layout.layout.dashboard'] = array(
+    'label' => t('Dasboard default layout'),
+    'group' => t('Configuration'),
+  );
+  $prefixes['dashboard.settings'] = array(
+    'label' => t('Dasboard settings'),
+    'group' => t('Configuration'),
+  );
+  return $prefixes;
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function dashboard_block_info() {
+  $blocks = array();
+
+  // Dashboard blocks.
+  $blocks['create'] = array(
+    'info' => t('Dashboard - Create content'),
+    'description' => t('Provides links to create new content'),
+    'class' => 'DashboardCreateBlock',
+  );
+  $blocks['menu'] = array(
+    'info' => t('Dashboard - Menus'),
+    'description' => t('Provides links to administer menus'),
+    'class' => 'DashboardMenuBlock',
+  );
+  $blocks['node_types'] = array(
+    'info' => t('Dashboard - Content types'),
+    'description' => t('Provides links to administer content types'),
+    'class' => 'DashboardContentTypesBlock',
+  );
+  $blocks['overview_content'] = array(
+    'info' => t('Dashboard - Content'),
+    'description' => t('Displays a summary of content statistics'),
+    'class' => 'DashboardOverviewContentBlock',
+  );
+  $blocks['overview_user'] = array(
+    'info' => t('Dashboard - User accounts'),
+    'description' => t('Displays a summary of user account statistics'),
+    'class' => 'DashboardOverviewUserBlock',
+  );
+  $blocks['taxonomy'] = array(
+    'info' => t('Dashboard - Categories'),
+    'description' => t('Provides links to administer taxonomy'),
+    'class' => 'DashboardTaxonomyBlock',
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_autoload_info().
+ */
+function dashboard_autoload_info() {
+  return array(
+    'DashboardCreateBlock' => 'dashboard_blocks/block.create.inc',
+    'DashboardMenuBlock' => 'dashboard_blocks/block.menus.inc',
+    'DashboardContentTypesBlock' => 'dashboard_blocks/block.node_types.inc',
+    'DashboardOverviewContentBlock' => 'dashboard_blocks/block.overview_content.inc',
+    'DashboardOverviewUserBlock' => 'dashboard_blocks/block.overview_user.inc',
+    'DashboardTaxonomyBlock' => 'dashboard_blocks/block.taxonomy.inc',
+  );
+}
+
+/**
+ * Implements hook_user_presave().
+ */
+function dashboard_user_presave($account) {
+  cache_clear_all('dashboard_user_overview', 'cache');
+}
+
+/**
+ * Implements hook_user_delete().
+ */
+function dashboard_user_delete($account) {
+  cache_clear_all('dashboard_user_overview', 'cache');
+}
+
+/**
+ * Implements hook_user_login().
+ * Redirects to the dashboard if configured.
+ */
+function dashboard_user_login(&$edit, $account) {
```

Comment:
There's a lot of nesting conditions going on here which makes it hard to understand. I don't know if Backdrop has a code style for this, but I've found this link helpful in making my conditionals easier to read: https://carlalexander.ca/mastering-php-conditionals/

Here's an alternative version (I haven't tested that it actually works):

```
function dashboard_user_login(&$edit, $account) {
  // Don't redirect if a destination is already set.
  if (!empty($_GET['destination'])) {
    return;
  }
  // Don't redirect for password reset.
  if (isset($_POST['form_id']) && $_POST['form_id'] == 'user_pass_reset') {
    return;
  }
  // Check dashboard redirect setting.
  if (config_get('dashboard.settings', 'dashboard_login_redirect') != 1) {
    return;
  }
  // Check for permission to use dashboard.
  if (!user_access('access dashboard', $account)) {
    return;
  }
  // Check for permission to access dashboard layout.
  $dashboard = layout_load('dashboard');
  if ($dashboard && ($dashboard->disabled || !$dashboard->checkAccess())) {
    return;
  }
  $_GET['destination'] = 'admin/dashboard';
}
```

Reply:
Fixed. (will push a new PR shortly)